modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java

```diff
@@ -639,6 +653,37 @@ private boolean initme() {
         return true;
     }
 
+    /**
+     * Use secure vault to secure password in JMS Message Store
+     * @param newParamValue
+     * @return
+     */
+    private String resolveSecureVaultExpressions(String newParamValue) {
+        Pattern vaultLookupPattern = Pattern.compile(secureVaultRegex);
+        Matcher lookupMatcher = vaultLookupPattern.matcher(newParamValue);
+        //setting value initially
+        if(lookupMatcher.find()) {
```

Comment:
What if lookupMatcher.find() == false? Should code handle that, or inform that in the log/UI?